### ISSUE TITLE
I have some trouble to reproduce result

### ISSUE THREAD

**Issue by LucasBoTang** [facebookresearch/neural-rewriter#7]
Hi authors,

I am trying to train for the JSP task with the default hyperparameters in code (64 batch size and 5e-5 learning rate), hyperparameters in your paper (128 batch size and 1e-4 learning rate) and other settings. After one epoch and more, I am afraid the loss and reward of my training are not converged. Since the training for one epoch takes a long time in my workstation, it is very frustrating.

Now I created a [colab](https://colab.research.google.com/drive/1LCItR-WOI8zv5Nls1SfXN-QGXEAufCTs). Could you please take a look and give me some help. I will appreciate it.

**Comment by Jungyhuk** [comment]
From my understanding of your colab and your email, at least some of the model checkpoints in your later training process should have a good performance if you evaluate it.

When you perform the training process, the rewriting traces are randomly sampled according to the learned distribution, and due to the randomness, they may not be optimal and even not good occasionally. Also, I manually increase the magnitude of the training losses in the dumped log file, so they may not seem stable by the values themselves.

To evaluate the model, you may adjust your command in the following way:

run_jsp.py --eval --load_model /path/to/ckpt --max_reduce_steps 100

The README includes the description of some flags above. For jsp with a larger size, it would be beneficial to allow a larger number of rewriting steps during evaluation to achieve better performance.

If you have more detailed questions, feel free to reply to our email thread for further clarification.

**Comment by LucasBoTang** [comment]
Thank you for your clarification.  I think now I know what I missed -- the reward you used for jsp task to the slowdown (In fact, it is _minus_ slow down so we can minimize the average slowdown), which confused me a lot previously because the logs and printout is not minus.

It might also be confusing for others since we expect the "reward" is something we want to maximize.

Thank you for your time again. I really appreciate!

**Comment by LucasBoTang** [comment]
Thank you for your clarification. I think now I know what I missed -- the reward you used for the jsp task is the slowdown (In fact, it is minus slowdown so we can minimize the average slowdown), which confused me a lot previously because the logs and printouts are not the minus one.

It might also be confusing for others since we expect the "reward" is something we want to maximize.

Thank you for your time again. I really appreciate!

**Comment by Jungyhuk** [comment]
Oh I see, yes your understanding is correct, sorry for the confusion!
